tests/_utils/kafka.py

```diff
@@ -0,0 +1,152 @@
+        partitions_assigned = len(_kafka_consumer.assignment()) > 0
+        if not partitions_assigned:
+            time.sleep(0.1)
+
+    assert partitions_assigned
+
+    _kafka_consumer.seek_to_beginning()
+
+    yield _kafka_consumer
+
+
+def _init_pytest_kafka() -> (bool, Exception):
+    """
+    Since the Kafka tests don't run by default, we will silently fail to initialize unless --run_kafka is enabled.
+    This should only be called once by the root conftest.py
+    """
+    global kafka_server, _kafka_consumer, zookeeper_proc  # pylint: disable=global-statement
+    try:
+        import pytest_kafka
+        os.environ['KAFKA_OPTS'] = "-Djava.net.preferIPv4Stack=True"
+        # Initialize pytest_kafka fixtures following the recomendations in:
+        # https://gitlab.com/karolinepauls/pytest-kafka/-/blob/master/README.rst
+        kafka_scripts = os.path.join(os.path.dirname(pytest_kafka.__file__), 'kafka/bin/')
+        if not os.path.exists(kafka_scripts):
+            # check the old location
+            kafka_scripts = os.path.join(os.path.dirname(os.path.dirname(pytest_kafka.__file__)), 'kafka/bin/')
+
+        kafka_bin = os.path.join(kafka_scripts, 'kafka-server-start.sh')
+        zookeeper_bin = os.path.join(kafka_scripts, 'zookeeper-server-start.sh')
+
+        for kafka_script in (kafka_bin, zookeeper_bin):
+            if not os.path.exists(kafka_script):
+                raise RuntimeError(f"Required Kafka script not found: {kafka_script}")
+
+        teardown_fn = partial(pytest_kafka.terminate, signal_fn=subprocess.Popen.kill)
+        zookeeper_proc = pytest_kafka.make_zookeeper_process(zookeeper_bin, teardown_fn=teardown_fn, scope='session')
+        kafka_server = pytest_kafka.make_kafka_server(kafka_bin,
+                                                      'zookeeper_proc',
+                                                      teardown_fn=teardown_fn,
+                                                      scope='session')
+        _kafka_consumer = pytest_kafka.make_kafka_consumer('kafka_server', scope='function')
+
+        return (True, None)
+    except Exception as e:
+        return (False, e)
+
+
+def write_data_to_kafka(bootstrap_servers: str,
+                        kafka_topic: str,
+                        data: list[str | dict],
+                        client_id: str = 'morpheus_unittest_writer') -> int:
+    """
+    Writes `data` into a given Kafka topic, emitting one message for each line int he file. Returning the number of
+    messages written
+    """
+    # pylint: disable=import-error
+    from kafka import KafkaProducer
+    num_records = 0
+    producer = KafkaProducer(bootstrap_servers=bootstrap_servers, client_id=client_id)
+    for row in data:
+        if isinstance(row, dict):
+            row = json.dumps(row)
+        producer.send(kafka_topic, row.encode('utf-8'))
+        num_records += 1
+
+    producer.flush()
+
+    assert num_records > 0
```

Comment:
Why are we summing `num_records`? Wouldn't that be the same as `len(data)`? Unless `producer.send` or `json.dumps` raises an error... in which case we wouldn't use the value anyway?

Reply:
I don't recall how we ended up with this code. I didn't modify this method, simply moved it from another module, however I believe you are correct num_records should be the same as `len(data)`